ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -75,6 +76,9 @@ public Http2NetworkClient(Http2ClientMetrics http2ClientMetrics, Http2ClientConf
         new StreamChannelInitializer());
     this.http2ClientMetrics = http2ClientMetrics;
     correlationIdInFlightToChannelMap = new ConcurrentHashMap<>();
+    Gauge<Long> nettyPendingTasks = () -> Http2Utils.getNumberOfPendingTasks(eventLoopGroup);
```

Comment:
Instead of exposing the registry, we usually add a method like `registerNettyPendingTasksGauge(Gauge<Long>)` to the metrics class to register the gauge but not require the other class to handle setting the metric name.

Same comment for `AmbryServer`